assets/js/components/setup/CompatibilityChecks.js

```diff
@@ -55,16 +50,11 @@ const checks = [
 			throw ERROR_INVALID_HOSTNAME;
 		}
 	},
-	// Generate and check for a Site Kit specific meta tag on the page to test for aggressive caching.
+	// Check for a Site Kit specific meta tag on the page to test for aggressive caching.
 	async () => {
-		const { token } = await API.set( 'core', 'site', 'setup-tag' );
-
-		const scrapedTag = await getExistingTag( 'setup' ).catch( () => {
-			throw ERROR_FETCH_FAIL;
-		} );
-
-		if ( token !== scrapedTag ) {
-			throw ERROR_TOKEN_MISMATCH;
+		const setupTag = await Data.dispatch( CORE_SITE ).checkForSetupTag();
```

Comment:
Ahh, what would `createCompatibilityChecks()` return? Passing a function to `useState` [does special stuff](https://medium.com/swlh/how-to-store-a-function-with-the-usestate-hook-in-react-8a88dd4eede1). But I think if `createCompatibilityChecks( registry )` returns an array that's fine. Maybe this was a typo and @aaemnnosttv meant:

```js
const [ ...compatibilityChecks ] = useState( createCompatibilityChecks( registry ) );
// or
const compatibilityChecks = useState( createCompatibilityChecks( registry ) );
```

I think that would work fine. 👍🏻 

At first I thought I also think putting the `useState` and even `useRegistry` into `useChecks` would be nicer. But that would make `useChecks` look like this:

```js
useChecks( createCompatibilityChecks, _registry ) => {
  let registry = useRegistry():
  registry = _registry || registry;
  const compatibilityChecks = useState( createCompatibilityChecks( registry ) );
}
```

Not sure if that's better. You could reduce the amount of boilerplate needed with that approach, but you'd need to pass the factory function to `useChecks` instead of the actual checks.

Dunno what you prefer—I like not having to do as much "setup" for the hook personally, but either would work.